test/test_library/unit/test_files.py

```diff
@@ -0,0 +1,511 @@
+        self.assertEquals(new_f.source_url, 'http://example.com')
+
+    # .merge tests
+    @unittest.skip('Where is Files().merge method? And why it is used in files.py?')
+    def test_adds_given_file_to_collection(self):
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+
+        f1 = FileFactory()
+        all_files.merge(f1, commit='collect')
+        self.assertEquals(len(all_files._collection), 1)
+        self.assertIn(f1.insertable_dict, all_files._collection)
+
+    @unittest.skip('Where is Files().merge method? And why it is used in files.py?')
+    def test_saves_given_file_to_db(self):
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+
+        self.assertEquals(len(all_files._collection), 0)
+        f1 = File(oid=1, path='/path1')
+        all_files.merge(f1)
+        self.assertEquals(len(all_files.all), 1)
+        self.assertIn(f1, all_files.all)
+
+    # .install_bundle tests
+    def test_saves_bundle_to_db(self):
+
+        # TODO: Create BundleFactory and use it here.
+        class FakeBundle(object):
+            database = fudge.Fake().has_attr(path='/path1')
+            identity = fudge.Fake().has_attr(vid='1')
+
+        bundle = FakeBundle()
+        source = 'http://example.com'
+
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+        new_f = all_files.install_bundle_file(bundle, source)
+
+        installed = all_files.all[0]
+        self.assertEquals(installed.path, bundle.database.path)
+        self.assertEquals(installed.ref, bundle.identity.vid)
+        self.assertEquals(installed.type_, Files.TYPE.BUNDLE)
+        self.assertEquals(installed.data, {})
+        self.assertEquals(installed.source_url, source)
+
+        # the same file returned
+        self.assertEquals(new_f.path, installed.path)
+        self.assertEquals(new_f.ref, installed.ref)
+        self.assertEquals(new_f.type_, installed.type_)
+
+    # .install_partition tests
+    @unittest.skip('What is partition.database?')
+    def test_saves_partition_to_db(self):
+
+        partition1 = PartitionFactory()
+        cache = fudge.Fake().has_attr(repo_id='1')
+
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+        new_f = all_files.install_partition_file(partition1, cache)
+
+        installed = all_files.all[0]
+        self.assertEquals(installed.path, partition1.database.path)
+        self.assertEquals(installed.group, cache.repo_id)
+        self.assertEquals(installed.ref, partition1.identity.vid)
+        self.assertEquals(installed.type_, Files.TYPE.PARTITION)
+        self.assertEquals(installed.data, {})
+        self.assertEquals(installed.source_url, None)
+        self.assertEquals(new_f, installed)
+
+    # .install_bundle_source tests
+    @unittest.skip('source can not be null.')
```

Comment:
Is 'source' unused parameter? https://github.com/nmb10/ambry/blob/issue-8/ambry/library/files.py#L233 .
TODO: Delete 'source' parameter?